chain/events/filter/index.go

```diff
@@ -108,6 +127,25 @@ type EventIndex struct {
 	stmtInsertEntry          *sql.Stmt
 	stmtRevertEventsInTipset *sql.Stmt
 	stmtRestoreEvent         *sql.Stmt
+	stmtInsertEventsSeen     *sql.Stmt
+	stmtRevertEventSeen      *sql.Stmt
+	stmtRestoreEventSeen     *sql.Stmt
+
+	mu           sync.Mutex
+	subIdCounter uint64
+	updateSubs   map[uint64]*updateSub
+}
+
+type updateSub struct {
+	ctx    context.Context
+	ch     chan *EventIndexUpdate
+	cancel context.CancelFunc
+}
+
+type EventIndexUpdate struct {
+	Height    uint64
```

Comment:
this should probably be `abi.ChainEpoch`, I see some casting going on that would be redundant here

Reply:
Done.